### PR TITLE
Fix many Disguise bugs

### PR DESCRIPTION
**Pull request by BuffelSaft**
Use an unused hitmarker flag to indicate that an HP change shouldn't break Disguise. 

While it seems a little wasteful to dedicate a flag to this, it's a simple, effective solution that makes fixing any edge cases trivial - just set the flag in the appropriate battle script and it'll work as expected.

This PR addresses #1407, #1566, #1626 and some other bugs that don't have issues. Namely:
- Mimikyu's HP draining moves break Disguise when they restore its HP
- HP restoring items break Disguise
- Basically anything that changes HP using gBattleMoveDamage breaks disguise (even if the value is negative).

### CODE REVIEW

**Review by DizzyEggg — APPROVED** [review]
That's a nice solution
